src/node/plugin.ts

```diff
@@ -23,38 +27,58 @@ require("module")._load = function (request: string, parent: object, isMain: boo
   return originalLoad.apply(this, [request.replace(/^code-server/, path.resolve(__dirname, "../..")), parent, isMain])
 }
 
+/**
+ * Load a plugin and run its activation function.
+ */
 const loadPlugin = async (pluginPath: string, httpServer: HttpServer, args: Args): Promise<void> => {
   try {
     const plugin: Plugin = require(pluginPath)
     plugin.activate(httpServer, args)
-    logger.debug("Loaded plugin", field("name", path.basename(pluginPath)))
+
+    logger.debug(
+      "Loaded plugin",
+      field("name", path.basename(pluginPath)),
+      field("version", require(path.join(pluginPath, "package.json")).version || "n/a"),
+    )
   } catch (error) {
-    if (error.code !== "MODULE_NOT_FOUND") {
-      logger.warn(error.message)
-    } else {
-      logger.error(error.message)
-    }
+    logger.error(error.message)
   }
 }
 
-const _loadPlugins = async (httpServer: HttpServer, args: Args): Promise<void> => {
-  const pluginPath = path.resolve(__dirname, "../../plugins")
-  const files = await util.promisify(fs.readdir)(pluginPath, {
-    withFileTypes: true,
-  })
-  await Promise.all(files.map((file) => loadPlugin(path.join(pluginPath, file.name), httpServer, args)))
-}
-
-export const loadPlugins = async (httpServer: HttpServer, args: Args): Promise<void> => {
+/**
+ * Load all plugins in the specified directory.
+ */
+const _loadPlugins = async (pluginDir: string, httpServer: HttpServer, args: Args): Promise<void> => {
   try {
-    await _loadPlugins(httpServer, args)
+    const files = await util.promisify(fs.readdir)(pluginDir, {
+      withFileTypes: true,
+    })
+    await Promise.all(files.map((file) => loadPlugin(path.join(pluginDir, file.name), httpServer, args)))
   } catch (error) {
     if (error.code !== "ENOENT") {
```

Comment:
Wouldn't loadPlugin never throw?

Reply:
Yup, it's the `readdir` part that might throw in this case.

Reply:
Gotchu!